src/scripts/components/widgets/modals/Modal.js

```diff
@@ -163,26 +168,42 @@ class Modal extends Component<Props, void> {
   }
 
   renderModal () {
-    const { modalContent } = this.props
-    switch (modalContent) {
+    const { modalName, modalProps } = this.props
+
+    let ModalComponent: any
+
+    switch (modalName) {
       case MODAL.STAKE:
-        return <ModalStake />
+        ModalComponent = ModalStake
+        break
       case MODAL.SECURE:
-        return <ModalSecure />
+        ModalComponent = ModalSecure
+        break
       case MODAL.GREAT_POWER:
-        return <ModalGreatPower />
+        ModalComponent = ModalGreatPower
+        break
       case MODAL.CREATE_PASSWORD:
-        return <ModalCreatePassword />
+        ModalComponent = ModalCreatePassword
+        break
       case MODAL.ASK_PASSWORD:
-        return <ModalAskPassword />
+        ModalComponent = ModalAskPassword
+        break
       case MODAL.SHOW_SEED:
-        return <ModalShowSeed />
+        ModalComponent = ModalShowSeed
+        break
       case MODAL.PROFILE:
-        return <ModalProfile />
+        ModalComponent = ModalProfile
+        break
       case MODAL.REWRITE_SEED:
-        return <ModalRewriteSeed />
+        ModalComponent = ModalRewriteSeed
+        break
       case MODAL.RESTORE_ACCOUNT:
-        return <ModalRestoreAccount />
+        ModalComponent = ModalRestoreAccount
+        break
+    }
+
+    if (ModalComponent) {
+      return <ModalComponent {...modalProps} />
```

Comment:
Now we can dispatch modal actions that include addl properties to pass to the modal component